### PR TITLE
Update to shapely v1.8.5 and add fix for gevent patching

### PR DESCRIPTION
**Pull request by olivia-hong**
Adds a fix for gevent monkeypatching on top of shapely v1.8.5
1.8.5 also supports Python 3.11 (upstream releases manylinux and macosx wheels).
Reverts breaking changes to `geos.py` from https://github.com/shapely/shapely/pull/1109, which switched shapely over to PEP 3135 (calling super() with no arguments).

Tested in multiple lyft repos
Without this fix, unit tests failed, but after I editable installed this branch, all unit tests passed for both repos.

### CODE REVIEW

File: shapely/__init__.py [hunk 1]
```diff
@@ -1 +1 @@
-__version__ = "1.8.5"
+__version__ = "1.8.5.post2"
```
Comment (olivia-hong) [inline shapely/__init__.py:1]: Note: 1.8.5.post1 exists upstream already

**Comment by aneeshusa** [comment]
We shouldn't be merging into the default branch, we should be using topic branches per our OSS topics. Happy to discuss tomorrow and will still review the contents but that popped out at me.

File: shapely/__init__.py [hunk 2]
```diff
@@ -1 +1 @@
-__version__ = "1.8.5.post1"
+__version__ = "1.8.5.post1.post1"
```
Comment (olivia-hong) [inline shapely/__init__.py]: Note that this look strange but is valid and done rather than "1.8.5.post2" in case upstream actually releases that version

Comment (olivia-hong) [reply shapely/__init__.py]: nvm this actually may be invalid as the `packaging` library doesn't recognize it

**Review by aneeshusa — APPROVED** [review]

File: shapely/__init__.py (hunk 2 above)
Comment (olivia-hong) [reply shapely/__init__.py]: going back to `.post2`

**Pull request merged**